### ISSUE TITLE
`default` target results in `Uncaught ReferenceError: require is not defined`

### ISSUE THREAD

**Issue by ericsoco** [parcel-bundler/parcel#4003]
# 🐛 bug report
As suggested by @mischnic [here](https://github.com/parcel-bundler/parcel/issues/3816#issuecomment-556261794), I'm targeting `default`. The build succeeds, but running the build in a browser results in `Uncaught ReferenceError: require is not defined`.

Other attempted targets (`main`, `browser`) result in a "External modules are not supported when building for browser" build error.

My intended environment is the browser, so perhaps I'm misusing the `default` target? Hard to tell from the docs. That said, if I should be using the `browser` target, I can't figure out how to get around the "External modules not supported" error.

## 🎛 Configuration (.babelrc, package.json, cli command)
All config is viewable in [the GH repo](https://github.com/ericsoco/fire-map/tree/aba3aa071dead1250a8b10989124114a4e3b7aba).

Also pasted below for reference.

`package.json`:
```js
  "default": "dist/index.html",
  "targets": {
    "default": {
      "publicUrl": "./"
    }
  }
```

`.parcelrc`:
```js
{
  "extends": "@parcel/config-default",
  "transforms": {
    "url:*": ["@parcel/transformer-raw"]
  }
}

```

`.babelrc`:
```js
{
	"presets": [
		"@babel/preset-env",
		"@babel/preset-react"
	],
  "plugins": [
    "@babel/plugin-proposal-optional-chaining",
    ["@babel/plugin-proposal-pipeline-operator", { "proposal": "minimal" }],
    "babel-plugin-styled-components",
    [
      "babel-plugin-import",
      {
        "libraryName": "@material-ui/core",
        "libraryDirectory": "",
        "camel2DashComponentName": false
      },
      "mui-core"
    ],
    [
      "babel-plugin-import",
      {
        "libraryName": "@material-ui/icons",
        "libraryDirectory": "",
        "camel2DashComponentName": false
      },
      "mui-icons"
    ]
  ]
}
```

## 🤔 Expected Behavior

The build should run in the browser.

## 😯 Current Behavior

Running the build in a browser results in `Uncaught ReferenceError: require is not defined`.

```
Uncaught ReferenceError: require is not defined
    at fire-map.512febdf.js:15
    at fire-map.512febdf.js:28
```

## 💁 Possible Solution

Maybe use a different target? See above notes about the "External modules not supported" error.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-alpha.3.2 |
| Node             | 10.16.0 |
| npm/Yarn         | yarn@1.19.2 |
| Operating System | OSX |

**Comment by mischnic** [comment]
(Instead of `default`, you can use any name. The only targets with a special meaning are `module`,`browser` and `main`, which don't bundle dependencies.)


The statement in question is
```
var process = require("process");
```

We should really disable the automatic polyfilling of that...

**Comment by ericsoco** [comment]
Yeah, I figured out that the two other `require`s were due to a `browser` field in the `package.json` of one of my dependencies. The package authors have since removed that field, and bumping my dep removed those `require`s.

However, at runtime I'm still unable to load external assets (via `axios`/XHR/fetch) that [I include like this](https://github.com/ericsoco/fire-map/blob/aba3aa071dead1250a8b10989124114a4e3b7aba/src/hooks/use-all-fires-for-year-request.js#L9):

```
import fires2010 from 'url:~/static/data/allPerimeters_California_2010.geojson';
```

These files are copied by `parcel build` into `dist` and given a hashed filename, but when the XHR call goes out the filename appears as `[Object object]`. Not sure how the hashed filenames are intended to be exposed to the bundled application, but looks like there's a parsing error somewhere. Should I open a new ticket for this?
